pymc/tests/test_distributions_moments.py

```diff
@@ -817,6 +818,34 @@ def test_mv_normal_moment(mu, cov, size, expected):
     assert_moment_is_expected(model, expected)
 
 
+@pytest.mark.parametrize(
+    "mu, size, expected",
+    [
+        (
+            np.array([1, 0, 3.0, 4]),
+            None,
+            np.array([1, 0, 3.0, 4]),
+        ),
+        (np.array([1, 0, 3.0, 4]), 6, np.full((6, 4), [1, 0, 3.0, 4])),
+        (np.array([1, 0, 3.0, 4]), (5, 3), np.full((5, 3, 4), [1, 0, 3.0, 4])),
+        (
+            np.array([[3.0, 5, 2, 1], [1, 4, 0.5, 9]]),
+            (4, 5),
+            np.full((4, 5, 2, 4), [[3.0, 5, 2, 1], [1, 4, 0.5, 9]]),
+        ),
+    ],
+)
+def test_car_moment(mu, size, expected):
+    W = np.array(
+        [[0.0, 1.0, 1.0, 0.0], [1.0, 0.0, 0.0, 1.0], [1.0, 0.0, 0.0, 1.0], [0.0, 1.0, 1.0, 0.0]]
+    )
+    tau = 2
+    alpha = 0.5
```

Comment:
Are tau and alpha constrained to be scalars?

Reply:
just noticed the class docstring allows them to be arrays. I just don't see it being tested anywhere in 
https://github.com/pymc-devs/pymc/blob/99ec0ff93362a6d6eba90cbad49a2338c353eccd/pymc/tests/test_distributions.py#L3162-L3242

and

https://github.com/pymc-devs/pymc/blob/99ec0ff93362a6d6eba90cbad49a2338c353eccd/pymc/tests/test_distributions_random.py#L2368-L2405

which is what this PR's tests are based on. I also don't remember any literature that treats tau is anything more than a scalar.

Reply:
If the random/logp methods work with non-scalar inputs for those parameters, we need `get_moment` to do the same. If they are not really supposed to work with those being non-scalar we don't need to, but we might have to open a new issue similar to https://github.com/pymc-devs/pymc/issues/5214

Reply:
#5214 was fixed in #5241 - the changeset was rather simple. @zoj613 do you want to add the enforcement of scalar `tau` and `alpha`, or add them as vectors to the test cases?

Reply:
According to https://github.com/pymc-devs/pymc/blob/b6f76e54485dcd376acbf188fdc9262ac40456c2/pymc/distributions/multivariate.py#L1999

it looks like tau and alpha are both scalars. The docstring seems wrong about allowing array input for those parameters.

Reply:
`ndims_params` indicates the minimum/base dimensions, not whether they can't be of higher dimensionality. Idelly we wouldn't impose any limitations, but the rng_fn/logp methods might not broadcast/handle higer values than the minimum case properly

Reply:
@ckrapu is the `tau` parameter meant to handle array input? I have not seen that case in literature. I also do not see that case being tested in the implementation unittests. I'm asking because it appears you submitted the PR for the distribution. Would you be able to clarify if we need to account for non-scalar tau?

Reply:
@zoj613 I think it's safe to go ahead and add that explicit limitation for the time being. We can always lift it later (and add tests). Do you want to include that in this PR?

Reply:
@ricardoV94  I think it's best that array tau be handled as a separate PR. I believe there is no good motivation to account for array tau given its lack of application in literature. Even the paper linked in the doc-string only mentions tau as an array in the case an MCAR model, which I believe is different from what is implemented by CAR.

Reply:
Lack of application in the literature shouldn't be a reason to restrict it, but instead not proper handling (or doubts of) on our end. I agree it can be done in a separate PR, would you be interested in doing that? I'll go ahead and merge this one after one last review